pkg/crypto/legacy_crypto.go

```diff
@@ -0,0 +1,115 @@
+import (
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite/ed25519signature2018"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdri"
+	"github.com/hyperledger/aries-framework-go/pkg/kms/legacykms"
+	"github.com/mr-tron/base58"
+)
+
+// LegacyCrypto is designed to work with the Aries framework's legacy signer and the current did:peer vdri
+// implementation. Example:
+//
+// {
+//     "@context": [
+//         "https://w3id.org/did/v1"
+//     ],
+//     "authentication": [
+//         ""
+//     ],
+//     "created": "2020-07-24T15:33:28.025521469-04:00",
+//     "id": "did:peer:1zQmNTB5T3NQHYcuhJkECgapwJGA5G8gT1eU1VH3cZMjT5AP",
+//     "publicKey": [
+//         {
+//             "controller": "#id",
+//             "id": "",
+//             "publicKeyBase58": "MsZt5z3DkJg28eis3SYaaoUDRrSdXknoYkqk5d8Tsge",
+//             "type": "Ed25519VerificationKey2018"
+//         }
+//     ],
+//     "service": [
+//         {
+//             "id": "#agent",
+//             "priority": 0,
+//             "recipientKeys": [
+//                 "MsZt5z3DkJg28eis3SYaaoUDRrSdXknoYkqk5d8Tsge"
+//             ],
+//             "routingKeys": null,
+//             "serviceEndpoint": "http://example.com/didcomm",
+//             "type": "did-communication"
+//         }
+//     ],
+//     "updated": "2020-07-24T15:33:28.025521469-04:00"
+// }.
+type LegacyCrypto struct {
+	ls      legacykms.Signer
+	vdriReg vdri.Registry
+}
+
+// NewLegacy returns new LegacyCrypto.
+func NewLegacy(s legacykms.Signer, v vdri.Registry) *LegacyCrypto {
+	return &LegacyCrypto{
+		ls:      s,
+		vdriReg: v,
+	}
+}
+
+// SignPresentation signs the presentation.
+func (l *LegacyCrypto) SignPresentation(vp *verifiable.Presentation, signingDID string) (*verifiable.Presentation, error) { //nolint: lll
+	verKey, err := l.getVerKey(signingDID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to sign presentation : %w", err)
+	}
+
+	s := &legacySigner{
+		verkey: verKey,
+		ks:     l.ls,
+	}
+
+	signingCtx := &verifiable.LinkedDataProofContext{
+		VerificationMethod:      verKey,
+		SignatureRepresentation: verifiable.SignatureJWS,
+		SignatureType:           Ed25519Signature2018,
+		Suite:                   ed25519signature2018.New(suite.WithSigner(s)),
+		Purpose:                 Authentication,
+	}
+
+	err = vp.AddLinkedDataProof(signingCtx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to signed presentation : %w", err)
+	}
+
+	return vp, nil
+}
+
+func (l *LegacyCrypto) getVerKey(signingDID string) (string, error) {
+	d, err := l.vdriReg.Resolve(signingDID)
+	if err != nil {
+		return "", fmt.Errorf("failed to resolve did : %w", err)
+	}
+
+	if len(d.PublicKey) == 0 {
+		return "", fmt.Errorf("no public keys found on did %s", signingDID)
+	}
+
+	return base58.Encode(d.PublicKey[0].Value), nil
```

Comment:
Shouldn't the format of verificationMethod be like "did#keyID" ?

Reply:
@rolsonquadras It isn't (yet).

Reply:
@llorllale I'm wondering how the receiver resolves the keys from DIDDoc and verify the signature ? Do we have any bdd tests where we are actually verifying the signature generated with this ?

Reply:
@rolsonquadras 

<details><summary>did:peer doc</summary>

```json
{                                                                                                                                                                                                                  
    "@context": [
        "https://w3id.org/did/v1"
    ],  
    "authentication": [
        ""
    ],  
    "created": "2020-07-24T17:08:31.370411561-04:00",
    "id": "did:peer:1zQmXJ88C99EXHjkpBEfaxKFeN9qnGadEZfQWs8SSH6rbVwy",
    "publicKey": [
        {
            "controller": "#id",
            "id": "", 
            "publicKeyBase58": "FMAHL95Quyt1WofUmKqWPYvjw3tnXKJ9MtWAmYfCg3M2",
            "type": "Ed25519VerificationKey2018"
        }
    ],  
    "service": [
        {
            "id": "#agent",
            "priority": 0,
            "recipientKeys": [
                "FMAHL95Quyt1WofUmKqWPYvjw3tnXKJ9MtWAmYfCg3M2"
            ],
            "routingKeys": null,
            "serviceEndpoint": "http://example.com/didcomm",
            "type": "did-communication"
        }
    ],  
    "updated": "2020-07-24T17:08:31.370411561-04:00"
}

```
</details>

<details><summary>signed VP</summary>

```json
{                                                                                                                                                                                                                  
    "@context": [
        "https://www.w3.org/2018/credentials/v1"
    ],  
    "proof": {
        "created": "2020-07-24T17:08:31.84389032-04:00",
        "jws": "eyJhbGciOiJFZERTQSIsImI2NCI6ZmFsc2UsImNyaXQiOlsiYjY0Il19..xy2NZy3TxBit6630-PkSuzxdRy4V1K8nq2WKX1-JG0SiLih1nXfY9mjNDCNhrE5xoz5LM7gkPojC6l7-tWR-Aw",
        "proofPurpose": "authentication",
        "type": "Ed25519Signature2018",
        "verificationMethod": "FMAHL95Quyt1WofUmKqWPYvjw3tnXKJ9MtWAmYfCg3M2"
    },  
    "type": "VerifiablePresentation",
    "verifiableCredential": [
        {
            "@context": [
                "https://www.w3.org/2018/credentials/v1",
                "https://www.w3.org/2018/credentials/examples/v1"
            ],
            "credentialSubject": {
                "degree": {
                    "degree": "MIT",
                    "type": "BachelorDegree"
                },
                "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
                "name": "Jayden Doe",
                "spouse": "did:example:c276e12ec21ebfeb1f712ebc6f1"
            },
            "id": "http://example.gov/credentials/ff98f978-588f-4eb0-b17b-60c18e1dac2c",
            "issuanceDate": "2020-03-16T22:37:26.544Z",
            "issuer": {
                "id": "did:web:vc.transmute.world",
                "name": "University"
            },
            "type": [
                "VerifiableCredential",
                "UniversityDegreeCredential"
            ]
        }
    ]   
}
```
</details>

The key is inlined in `VerificationMethod`.

Does the framework have an API to validate proofs? I assumed `verifiable.ParsePresentation()` did it.